### PR TITLE
fix(buildinfo): go version on windows

### PR DESCRIPTION
**Pull request by qianlongzt**
windows have colon in filename

image

### CODE REVIEW

File: internal/pkg/mod/print.go [hunk 1]
```diff
@@ -38,7 +38,10 @@ func (pkg *Pkg) Print() {
 
 func (pkg *Pkg) printFilename() {
 	if pkg.BuildInfo != nil && pkg.BuildInfo.Filename != "" {
-		printKeyValue("File", strings.TrimPrefix(pkg.BuildInfo.Filename, config.GOBIN+"/"))
+		name := strings.TrimPrefix(pkg.BuildInfo.Filename, config.GOBIN)
+		// window start with \
+		name = strings.TrimLeft(name, "/\\")
+		printKeyValue("File", name)
```
Comment (burik666) [inline internal/pkg/mod/print.go]:
```suggestion
		printKeyValue("File", strings.TrimPrefix(pkg.BuildInfo.Filename, config.GOBIN+string(os.PathSeparator)))
```

File: go.mod [hunk 2]
```diff
@@ -1,18 +1,19 @@
 module github.com/burik666/gobin
 
-go 1.17
+go 1.22.0
+
+toolchain go1.23.3
```
Comment (burik666) [inline go.mod]: `toolchain` directive is not needed here.

**Review by burik666 — COMMENTED** [review]
Thank you for your contribution.

File: go.mod (hunk 2 above)
Comment (qianlongzt) [reply go.mod]: do you mean upgrade go line to go 1.23.3?

Comment (burik666) [reply go.mod]: I mean that the toolchain directive should be removed.
```suggestion
go 1.22.0
```

Comment (qianlongzt) [reply go.mod]: ok

**Pull request merged**